icx/src/main.rs

```diff
@@ -251,16 +263,28 @@ fn print_idl_blob(
     Ok(())
 }
 
-async fn fetch_root_key_from_non_ic(agent: &Agent, replica: &str) -> Result<()> {
-    let normalized_replica = replica.strip_suffix('/').unwrap_or(replica);
-    if normalized_replica != DEFAULT_IC_GATEWAY {
-        agent
-            .fetch_root_key()
-            .await
-            .context("Failed to fetch root key from replica")?;
+async fn fetch_root_key_from_non_ic(agent: &Agent, replica: &Url) -> Result<()> {
+    if is_mainnet(replica) {
```

Comment:
Shouldn't this logic be the opposite?  We should never fetch the root key on mainnet, and should always fetch it elsewhere.
```suggestion
    if !is_mainnet(replica) {
```

Reply:
We could also consider the approach that icx-proxy takes: adding a `--fetch-root-key` parameter.

https://github.com/dfinity/ic/blob/master/rs/boundary_node/icx_proxy/src/main.rs#L83